### PR TITLE
FT: Support multiple replication storage classes

### PR DESCRIPTION
**Pull request by bennettbuchanan**
Depends on https://github.com/scality/S3/pull/1066.

### CODE REVIEW

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine 
SCALITY_S3_BRANCH=ft/S3C-1144/replication-config-multiple-storage-classes
try

File: lib/models/ReplicationConfiguration.js [hunk 1]
```diff
@@ -281,30 +280,28 @@ class ReplicationConfiguration {
...
+                validStorageClasses.includes(storageClass)) {
+                this._hasScalityDestination = true;
+                return true;
             }
-            this._isExternalLocation = true;
-        }
-        if (!replicationEndpoints.includes(storageClass) &&
-            !locationConstraints.includes(storageClass) &&
-            !validStorageClasses.includes(storageClass)) {
+            if (validStorageClasses.includes(storageClass)) {
```
Comment (philipyoo) [inline lib/models/ReplicationConfiguration.js]: i think the above `if` statement will always catch this condition

Comment (bennettbuchanan) [reply lib/models/ReplicationConfiguration.js]: This condition is for AWS compatible storage classes (e.g., 'STANDARD').

Comment (bennettbuchanan) [reply lib/models/ReplicationConfiguration.js]: Nice catch! Will update this.

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by rahulreddy** [comment]
@ironman-machine r=2c1bf72cc65da380a8a01fb74f27e16fd1ca9ecf

**Comment by rahulreddy** [comment]
@ironman-machine r-

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by rahulreddy** [comment]
@ironman-machine r=2c1bf72cc65da380a8a01fb74f27e16fd1ca9ecf

**Comment by rahulreddy** [comment]
@ironman-machine r- r=2c1bf72cc65da380a8a01fb74f27e16fd1ca9ecf

**Comment by rahulreddy** [comment]
@ironman-machine r- r=2c1bf72cc65da380a8a01fb74f27e16fd1ca9ecf

**Comment by rahulreddy** [comment]
@ironman-machine r- try

**Comment by rahulreddy** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine 
SCALITY_INTEGRATION_BRANCH=ft/S3C-1144/add-default-replication-endpoint
try

**Comment by bennettbuchanan** [comment]
@ironman-machine
SCALITY_BACKBEAT_BRANCH=ft/S3C-1144/update-populator-config-with-default
try

**Comment by bennettbuchanan** [comment]
E2E passing with these dependent branches:
http://ci.ironmann.io/gh/scality/Integration/18582#build-parameters

**Pull request merged**